### PR TITLE
updated auth params to include scope

### PR DESCRIPTION
**Pull request by acaruso-oddball**
**Note**: Delete the description statements, complete each step. **None are optional**, but can be justified as to why they cannot be completed as written. Provide known gaps to testing that may raise the risk of merging to production.

## Are you removing, renaming or moving a folder in this PR?
- [x] No, I'm not changing any folders (skip to Summary and delete the rest of this section)
- [ ] Yes, I'm removing, renaming or moving a folder

If the folder you changed contains a `manifest.json`, search for its `entryName` in the content-build [registry.json](https://github.com/department-of-veterans-affairs/content-build/blob/main/src/applications/registry.json) (the `entryName` there will match).

If an entry for this folder exists in content-build and you are:
1. **Deleting a folder**:
   1. First search `vets-website` for _all_ instances of the `entryName` in your `manifest.json` and remove them in a separate PR. Look particularly for references in `src/applications/static-pages/static-pages-entry.js` and `src/platform/forms/constants.js`. _**If you do not do this, other applications will break!**_
      - _Add the link to your merged vets-website PR here_
   2. Then, Delete the application entry in [registry.json](https://github.com/department-of-veterans-affairs/content-build/blob/main/src/applications/registry.json) and merge that PR **before** this one
      - _Add the link to your merged content-build PR here_

2. **Renaming or moving a folder**: Update the entry in the [registry.json](https://github.com/department-of-veterans-affairs/content-build/blob/main/src/applications/registry.json), but do not merge it until your vets-website changes here are merged. The content-build PR must be merged immediately after your vets-website change is merged in to avoid CI errors with content-build (and Tugboat).

### :warning: TeamSites :warning:
Examples of a TeamSite: https://va.gov/health and https://benefits.va.gov/benefits/. This scenario is also referred to as the "injected" header and footer. You can reach out in the `#sitewide-public-websites` Slack channel for questions.

Did you change site-wide styles, platform utilities or other infrastructure?
- [x] No
- [ ] Yes, and I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#that-sounds-normal-so-whats-the-proxy-all-about) to test the injected header scenario

## Summary
This update refactors the sessionTypeUrl and createOAuthRequest functions to support the inclusion of the scope query parameter specifically for mobile OAuth requests, enabling single sign-on (SSO) between VA.gov and the VA mobile app. The scope parameter is now available for all mobile applications (vamobile and vaoccmobile) and is omitted for non-mobile applications to maintain existing functionality. The code changes ensure that scope is only included in the query parameters when it has a defined value and when the request is coming from a mobile application. 

## Related issue(s)

[Jira 381](https://jira.devops.va.gov/browse/VI-381?jql=project%20%3D%20VI%20AND%20resolution%20%3D%20Unresolved%20ORDER%20BY%20priority%20DESC%2C%20updated%20DESC)

## Testing done

- Unit Tests: Added and updated unit tests to cover scenarios for mobile and non-mobile OAuth requests, ensuring that the scope parameter is correctly handled.
- Manual Tests: Conducted manual testing to verify that OAuth URLs are correctly generated in various scenarios, both with and without the scope parameter.

## Screenshots

_Note: This field is mandatory for UI changes (non-component work should NOT have screenshots)._

|         | Before | After |
| ------- | ------ | ----- |
| Mobile  |        |       |
| Desktop |        |       |

## What areas of the site does it impact?

*(Describe what parts of the site are impacted **if** code touched other areas)*

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [ ] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [x] Browser console contains no warnings or errors.
- [x] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [x] Did you login to a local build and verify all authenticated routes work as expected with a test user

## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

**Comment by bosawt** [comment]
Tested by navigating here, without `scope=device_sso`:
```
localhost:3001/?next=loginModal&oauth=true&application=vamobile&code_challenge=1BUpxy37SoIPmKw96wbd6MDcvayOYm3ptT-zbe6L_zM%3D
```
(Redirected to vamobile callback, as expected, manually used the code to call `/token` endpoint, got back):
```
{"data":{"refresh_token":"v1:insecure+data+A6ZXlKMWMyVnlYM1YxYVdRaU9pSXhPV1psTlRGaE5DMW1ZVEJsTFRReE9Ua3RPRE5rTnkwellXVXhNMkUwWkRCaVpEQWlMQ0oxZFdsa0lqb2lOVE5tTVdFNU1USXROelF6TmkwME56WXpMV0k0WlRjdE5UQXpNemt3WkdWaU1ETTJJaXdpYzJWemMybHZibDlvWVc1a2JHVWlPaUpoTXpnNU5UWTBaUzFrWlRBd0xUUTJNVEl0T0RZNU5TMDBZVEUxTkRJek56aGhZbUVpTENKd1lYSmxiblJmY21WbWNtVnphRjkwYjJ0bGJsOW9ZWE5vSWpvaU9ESmxaakl6TkRRMlltSXlOR0ZoWTJFMllqazNaVEU0WkdJeU4yVXhOVFl5WkRnMU9USTBaakEyWm1NM016RmxNR1UxTXpOallqazRZemRpWW1VNU1pSXNJbUZ1ZEdsZlkzTnlabDkwYjJ0bGJpSTZJbVZoTVRObE1qQTFaVGhoTWpJNVpXRmpNekkwTURnd05UVTBPV0UxWldNNElpd2libTl1WTJVaU9pSm1ZakkzTWpnNU5qTTRaVGd6TVRneE5qZ3dZVE5rTWpZNE1USmpNR05sWWlJc0luWmxjbk5wYjI0aU9pSldNQ0lzSW5aaGJHbGtZWFJwYjI1ZlkyOXVkR1Y0ZENJNmJuVnNiQ3dpWlhKeWIzSnpJanA3ZlgwPQ==.fb27289638e83181680a3d26812c0ceb.V0","access_token":"eyJhbGciOiJSUzI1NiJ9.eyJpc3MiOiJ2YS5nb3Ygc2lnbiBpbiIsImF1ZCI6WyJ2YW1vYmlsZSIsInZhd2ViIiwidmFtb2NrIiwic2FtcGxlX2NsaWVudF93ZWIiXSwiY2xpZW50X2lkIjoidmFtb2JpbGUiLCJqdGkiOiJlNGI1MWI4MS03ZjRhLTQyN2YtOWQxMy05YmYyYWE1NDFiYmQiLCJzdWIiOiIxOWZlNTFhNC1mYTBlLTQxOTktODNkNy0zYWUxM2E0ZDBiZDAiLCJleHAiOjE3MjQ5NTMzMjAsImlhdCI6MTcyNDk1MTUyMCwic2Vzc2lvbl9oYW5kbGUiOiJhMzg5NTY0ZS1kZTAwLTQ2MTItODY5NS00YTE1NDIzNzhhYmEiLCJyZWZyZXNoX3Rva2VuX2hhc2giOiJlMDFmMmJhZTlkNjAyYWZlYTExMzA4OTAxZWIzOWY0NTk4NjM0NjdjNjYxOTk0OTNiYzliZWQzOGM2MzM2ZTM5IiwiZGV2aWNlX3NlY3JldF9oYXNoIjpudWxsLCJwYXJlbnRfcmVmcmVzaF90b2tlbl9oYXNoIjoiODJlZjIzNDQ2YmIyNGFhY2E2Yjk3ZTE4ZGIyN2UxNTYyZDg1OTI0ZjA2ZmM3MzFlMGU1MzNjYjk4YzdiYmU5MiIsImFudGlfY3NyZl90b2tlbiI6ImVhMTNlMjA1ZThhMjI5ZWFjMzI0MDgwNTU0OWE1ZWM4IiwibGFzdF9yZWdlbmVyYXRpb25fdGltZSI6MTcyNDk1MTUyMCwidmVyc2lvbiI6IlYwIiwidXNlcl9hdHRyaWJ1dGVzIjp7fX0.Pc9fWdMJzL1BmD4D4puuMquovlyal8G_6K2Zy7uB4Q5NQGUDlvR-7oME2FKU-KxWixAhF6VywKCArYabMuSdOnVTly_PQtQMH64ibDkRSlkg6a-PM3Q1PumkWmNQZ_WDK8IDkd181X2KImNPVwr0_z5IRlmhVwu6tPhlIvqk4ni_QCeDZoiqaQ5z2D1UvvVTCEhIkTNbJfVp-yN4PRj39w9w1WhS6dNrotCKamxDkrP1aAFLdt3iZLMkNd9TAbqLzRIlP87FW6bIgZqYE2_-rhcyYNwz12k0P_MU2oxMkaC8j7pGneNZAdcM2u6VcAh6arYanE8n9Sllc5u_KESl6A"}}
```
Refresh token, access token, as expected

Next, navigated here, with `scope=device_sso`:
```
localhost:3001/?next=loginModal&oauth=true&application=vamobile&scope=device_sso&code_challenge=1BUpxy37SoIPmKw96wbd6MDcvayOYm3ptT-zbe6L_zM%3D
```

Followed same process, token response:
```
{"data":{"refresh_token":"v1:insecure+data+A6ZXlKMWMyVnlYM1YxYVdRaU9pSXhPV1psTlRGaE5DMW1ZVEJsTFRReE9Ua3RPRE5rTnkwellXVXhNMkUwWkRCaVpEQWlMQ0oxZFdsa0lqb2lOekEzTUdWaU56SXRNelJrT1MwME1UZGlMV0ZtWlRJdE5XWmpNMkUxWldSak1XUTRJaXdpYzJWemMybHZibDlvWVc1a2JHVWlPaUkxTTJFME9UTm1aUzA0TkdRMExUUTVNVGt0T0RJME5TMDBZbUV4WTJRNU5tTTJPRGNpTENKd1lYSmxiblJmY21WbWNtVnphRjkwYjJ0bGJsOW9ZWE5vSWpvaVl6UTFOall5T0RNeE16a3lOV1l3TlRJMllUZ3haVFJsTWpReFpUWTJaRGhoWVdObFptWmlaakE1TkRnME9XWXpOamczWTJJeE9EQmtNRGsyTVdRNVl5SXNJbUZ1ZEdsZlkzTnlabDkwYjJ0bGJpSTZJamRsTURWbE1qRTJZalU1TlRObE5tRmxZMkZtTVRnNU9ETTNNVE5sT1Rka0lpd2libTl1WTJVaU9pSm1aVE0xTkRVM05URXpPVFpsTkdZMU1EZGhNVFl4T1RoaVlqVXhZalF6WmlJc0luWmxjbk5wYjI0aU9pSldNQ0lzSW5aaGJHbGtZWFJwYjI1ZlkyOXVkR1Y0ZENJNmJuVnNiQ3dpWlhKeWIzSnpJanA3ZlgwPQ==.fe3545751396e4f507a16198bb51b43f.V0","access_token":"eyJhbGciOiJSUzI1NiJ9.eyJpc3MiOiJ2YS5nb3Ygc2lnbiBpbiIsImF1ZCI6WyJ2YW1vYmlsZSIsInZhd2ViIiwidmFtb2NrIiwic2FtcGxlX2NsaWVudF93ZWIiXSwiY2xpZW50X2lkIjoidmFtb2JpbGUiLCJqdGkiOiI2YTVhMjlmZi0zMWJmLTQ4ODctYmY0Zi1lMjA4YjM0YjRhYWUiLCJzdWIiOiIxOWZlNTFhNC1mYTBlLTQxOTktODNkNy0zYWUxM2E0ZDBiZDAiLCJleHAiOjE3MjQ5NTM1MTAsImlhdCI6MTcyNDk1MTcxMCwic2Vzc2lvbl9oYW5kbGUiOiI1M2E0OTNmZS04NGQ0LTQ5MTktODI0NS00YmExY2Q5NmM2ODciLCJyZWZyZXNoX3Rva2VuX2hhc2giOiIzMjcyODA4YmY1M2I4NDM5MDJlZDE0ZTVlOWUxYzZiNzZiNmE2MGI5MzExYzNlMTE0ZTI4ZTc1OWY5Y2ZmOTE5IiwiZGV2aWNlX3NlY3JldF9oYXNoIjoiOWZlZmNlYzMyNzEyOWNkMjY0YzdlN2ZhODkwZjcyNTM5MGIzODBiZWIzZTEzNzQ3YzJmY2I1ZjczNGM0YmI3NiIsInBhcmVudF9yZWZyZXNoX3Rva2VuX2hhc2giOiJjNDU2NjI4MzEzOTI1ZjA1MjZhODFlNGUyNDFlNjZkOGFhY2VmZmJmMDk0ODQ5ZjM2ODdjYjE4MGQwOTYxZDljIiwiYW50aV9jc3JmX3Rva2VuIjoiN2UwNWUyMTZiNTk1M2U2YWVjYWYxODk4MzcxM2U5N2QiLCJsYXN0X3JlZ2VuZXJhdGlvbl90aW1lIjoxNzI0OTUxNzEwLCJ2ZXJzaW9uIjoiVjAiLCJ1c2VyX2F0dHJpYnV0ZXMiOnt9fQ.UwPbflCrimJ2qQj8RbUY2z0y9k7nz3Dc2kIkq9MYmlU26wOqNQbHufvh4vQ9DPgiNpUuhplNdflBTQFcGsHh6Y76QUOUP6GiRpS1XGM54mB_pk1wnaNbAYrD7d_yreZbzDMjYStZ8q6DfQpGrrfZTBlQBC-ra8oIWW584OpJ4X7swle5Ig5GlIQVpUG6CNP5C5TIw9pOLdLMV6wyRP07IxWHqitixMQa_RvyEzWmA2jQnapPXH6YD9UDdkhZ5_lp9g4bynwxAmLQssK0icSKQXaUn_U8vLLpfYMDElmjR_4J1647YKJUuh5-A0ccKzNoq8pCDF8e02LWHuVp2h8cQQ","device_secret":"cfa9296c971f515246127598c4e5fac7"}}
```
Refresh token, access token, and device_secret, as expected

File: src/platform/user/authentication/utilities.js [hunk 1]
```diff
@@ -232,6 +233,7 @@ export function sessionTypeUrl({
       : {};
 
   if (useOAuth && (isLogin || isSignup)) {
+    const validScope = scope || undefined;
```
Comment (bosawt) [inline src/platform/user/authentication/utilities.js:236]: I don't think we need the `|| undefined` part of this var definition, won't it be undefined anyways if `scope` isn't defined?

Comment (acaruso-oddball) [reply src/platform/user/authentication/utilities.js:236]: Makes sense, updated.

**Review by ianmagenta — DISMISSED** [review]
Looks good overall! Just one small change. There are some required tests that are failing, but those should hopefully resolve through merges with main.

File: src/platform/user/authentication/utilities.js (hunk 1 above)
Comment (acaruso-oddball) [reply src/platform/user/authentication/utilities.js]: thx looks like the update didn't push at that time updated.

**Review by ianmagenta — DISMISSED** [review]
LGTM!

File: src/platform/user/authentication/utilities.js [hunk 2]
```diff
@@ -242,6 +243,8 @@ export function sessionTypeUrl({
         codeChallenge,
         codeChallengeMethod,
         ...(gaClientId && { gaClientId }),
+        // ...(validScope && { scope: validScope }),
```
Comment (bosawt) [inline src/platform/user/authentication/utilities.js:246]: Do we need this comment? Why not remove it?

**Review by bosawt — APPROVED** [review]

**Pull request merged**